src/datachain/sql/types.py

```diff
@@ -445,6 +446,18 @@ def string(self, value):
         return value
 
     def boolean(self, value):
+        if value is None or isinstance(value, bool):
+            return value
+
+        if isinstance(value, numbers.Integral):
+            return bool(value)
+        if isinstance(value, str):
+            normalized = value.strip().lower()
+            if normalized in {"true", "t", "yes", "y", "1"}:
```

Comment:
What if sometimes user really wanted a value named yes in string format? Wouldn't that be unexpected for user if some rows with value yes was converted to boolean under assumption?

Reply:
`read_csv` is using Arrow and automatically derives the type. It has separate `column_types` parameter to override that:

> Dictionary of column names and their corresponding types. It is passed to CSV reader and for each column specified type auto inference is disabled.